go/libraries/doltcore/sqle/commit_diff_table.go

```diff
@@ -0,0 +1,397 @@
+	if dt.requiredFilterErr != nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), dt.requiredFilterErr)
+	} else if dt.toCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneToCommit)
+	} else if dt.fromCommitFilter == nil {
+		return nil, fmt.Errorf("error querying table %s: %w", dt.Name(), ErrExactlyOneFromCommit)
+	}
+
+	toRoot, toName, toDate, err := dt.rootValForFilter(ctx, dt.toCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromRoot, fromName, fromDate, err := dt.rootValForFilter(ctx, dt.fromCommitFilter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	toTable, _, err := toRoot.GetTable(ctx, dt.name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromTable, _, err := fromRoot.GetTable(ctx, dt.name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSliceOfPartitionsItr([]sql.Partition{diffPartition{
+		to:       toTable,
+		from:     fromTable,
+		toName:   toName,
+		fromName: fromName,
+		toDate:   toDate,
+		fromDate: fromDate,
+	}}), nil
+}
+
+func (dt *CommitDiffTable) rootValForFilter(ctx *sql.Context, eqFilter *expression.Equals) (*doltdb.RootValue, string, *types.Timestamp, error) {
+	var hashStr string
+	var ok bool
+	if _, isGetField := eqFilter.BinaryExpression.Left.(*expression.GetField); !isGetField {
+		val, err := eqFilter.BinaryExpression.Left.Eval(ctx, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		if hashStr, ok = val.(string); !ok {
+			return nil, "", nil, fmt.Errorf("received '%v' when expecting commit hash string", val)
+		}
+	} else if _, isGetField = eqFilter.BinaryExpression.Right.(*expression.GetField); !isGetField {
+		val, err := eqFilter.BinaryExpression.Right.Eval(ctx, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		if hashStr, ok = val.(string); !ok {
+			return nil, "", nil, fmt.Errorf("received '%v' when expecting commit hash string", val)
+		}
+	}
+
+	var root *doltdb.RootValue
+	var commitTime *types.Timestamp
+	if strings.ToLower(hashStr) == "working" {
+		root = dt.workingRoot
+	} else {
+		cs, err := doltdb.NewCommitSpec(hashStr)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		cm, err := dt.ddb.Resolve(ctx, cs, nil)
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		root, err = cm.GetRootValue()
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		meta, err := cm.GetCommitMeta()
+
+		if err != nil {
+			return nil, "", nil, err
+		}
+
+		t := meta.Time()
+		commitTime = (*types.Timestamp)(&t)
+	}
+
+	return root, hashStr, commitTime, nil
+}
+
+// HandledFilters returns the list of filters that will be handled by the table itself
+func (dt *CommitDiffTable) HandledFilters(filters []sql.Expression) []sql.Expression {
+	var remaining []sql.Expression
+	for _, filter := range filters {
+		isCommitFilter := false
+
+		if eqFilter, isEquality := filter.(*expression.Equals); isEquality {
+			sql.Inspect(filter, func(e sql.Expression) (cont bool) {
```

Comment:
This isn't quite right: you can't use sql.Inspect here because it walks the entire expression tree, and you really just want the top level. You need to inspect the left and right of the equals expression directly. One of them must be a GetField to be handled.

Reply:
You can use the same `left, right = right, left` swapping trick here as well

Reply:
The way you have it now will catch expressions like `concat(from_commit, "abc")`, which will yield incorrect results with your logic below